doc/examples/applications/plot_cornea_spot_inpainting.py

```diff
@@ -0,0 +1,255 @@
+# as well as the ``offset`` (shifting the neighborhood's weighted mean).
+# Let us try two different values for ``block_size``:
+
+thresh_1 = filters.threshold_local(image_avg, block_size=21, offset=15)
+thresh_2 = filters.threshold_local(image_avg, block_size=43, offset=15)
+
+mask_1 = image_avg < thresh_1
+mask_2 = image_avg < thresh_2
+
+#####################################################################
+# Let us define a convenience function to display two plots side by side, so
+# it is easier for us to compare them:
+
+def plot_comparison(plot1, plot2, title1, title2):
+    fig, (ax1, ax2) = plt.subplots(
+        ncols=2,
+        figsize=(12, 6),
+        sharex=True,
+        sharey=True
+    )
+    ax1.imshow(plot1, cmap='gray')
+    ax1.set_title(title1)
+    ax2.imshow(plot2, cmap='gray')
+    ax2.set_title(title2)
+
+plot_comparison(mask_1, mask_2, "block_size = 21", "block_size = 43")
+
+#####################################################################
+# The "dark spots" appear to be more distinct in the second mask, i.e., the
+# one resulting from using the larger ``block_size`` value.
+# Regarding the value of the offset parameter, we noticed that increasing it
+# instead of keeping its default zero value, would yield a more uniform
+# background, hence letting the objects of interest stand out more visibly.
+# Indeed:
+
+thresh_0 = filters.threshold_local(image_avg, block_size=43)
+
+mask_0 = image_avg < thresh_0
+
+plot_comparison(mask_0, mask_2, "No offset", "offset = 15")
+
+#####################################################################
+# Remove fine-grained features
+# ============================
+# We use morphological filters to sharpen the mask and focus on the dark
+# spots. The two fundamental morphological operators are *dilation* and
+# *erosion*, where dilation (resp. erosion) sets the pixel to the brightest
+# (resp. darkest) value of the neighborhood defined by a structuring element
+# (footprint).
+#
+# Here, we use the ``diamond`` function from the ``morphology`` module to
+# create a diamond-shaped footprint.
+# An erosion followed by a dilation is called an *opening*.
+# First, we apply an opening filter, in order to remove small objects and thin
+# lines, while preserving the shape and size of larger objects.
+
+footprint = morphology.diamond(1)
+mask_open = morphology.opening(mask_2, footprint)
+plot_comparison(mask_2, mask_open, "mask before", "after opening")
+
+#####################################################################
+# Since "opening" an image starts with an erosion operation, bright regions
+# which are smaller than the structuring element have been removed.
+# Let us try now with a larger footprint:
+
+footprint = morphology.diamond(3)
+mask_open = morphology.opening(mask_2, footprint)
+plot_comparison(mask_2, mask_open, "mask before", "after opening")
+
+#####################################################################
```

Comment:
```suggestion
#####################################################################
# We can see that, with respect to the previous comparison plot, fewer spots
# are retained in the mask: The smaller ones have been filtered out, leaving
# only the larger ones.
```
@lagru @decorouz this suggestion follows from https://github.com/scikit-image/scikit-image/pull/6853#discussion_r1226429325.

Reply:
> The smaller ones have been filtered out, leaving only the larger ones.

This is basically true for every opening operation. What I meant in https://github.com/scikit-image/scikit-image/pull/6853#discussion_r1225823966, was to highlight how the footprint size affects this operation. That's only hinted at here, if at all.